cmd/openfga/main_test.go

```diff
@@ -569,7 +569,130 @@ func connect(t *testing.T, tester OpenFGATester) *grpc.ClientConn {
 }
 
 func GRPCWriteTest(t *testing.T, tester OpenFGATester) {
+	type output struct {
+		errorCode    codes.Code
+		errorMessage string
+	}
+
+	conn := connect(t, tester)
+	defer conn.Close()
+
+	client := openfgav1.NewOpenFGAServiceClient(conn)
+	resp, err := client.CreateStore(context.Background(), &openfgav1.CreateStoreRequest{
+		Name: "openfga-demo",
+	})
+	require.NoError(t, err)
+	storeID := resp.GetId()
+
+	model := parser.MustTransformDSLToProto(`model
+	schema 1.1
+type user
+
+type document
+  relations
+	define viewer: [user]
+`)
+
+	writeModelResp, err := client.WriteAuthorizationModel(context.Background(), &openfgav1.WriteAuthorizationModelRequest{
+		StoreId:         storeID,
+		TypeDefinitions: model.GetTypeDefinitions(),
+		Conditions:      model.GetConditions(),
+		SchemaVersion:   typesystem.SchemaVersion1_1,
+	})
+	require.NoError(t, err)
+	require.NotPanics(t, func() { ulid.MustParse(writeModelResp.GetAuthorizationModelId()) })
+	modelID := writeModelResp.GetAuthorizationModelId()
+
+	tests := []struct {
+		name   string
+		input  *openfgav1.WriteRequest
+		output output
+	}{
+		{
+			name: "happy-path",
+			input: &openfgav1.WriteRequest{
+				StoreId:              storeID,
+				AuthorizationModelId: modelID,
+				Writes: &openfgav1.WriteRequestWrites{
+					TupleKeys: []*openfgav1.TupleKey{
+						{Object: "document:1", Relation: "viewer", User: "user:jon"},
+					},
+				},
+			},
+			output: output{
+				errorCode: codes.OK,
+			},
+		},
+		{
+			name: "invalid-store-id",
+			input: &openfgav1.WriteRequest{
+				StoreId:              "invalid-store-id",
+				AuthorizationModelId: modelID,
+				Writes:               &openfgav1.WriteRequestWrites{},
+			},
+			output: output{
+				errorCode:    codes.InvalidArgument,
+				errorMessage: `value does not match regex pattern "^[ABCDEFGHJKMNPQRSTVWXYZ0-9]{26}$`,
+			},
+		},
+		{
+			name: "invalid-model-id",
+			input: &openfgav1.WriteRequest{
+				StoreId:              storeID,
+				AuthorizationModelId: "invalid-model-id",
+				Writes:               &openfgav1.WriteRequestWrites{},
+			},
+			output: output{
+				errorCode:    codes.InvalidArgument,
+				errorMessage: `value does not match regex pattern "^[ABCDEFGHJKMNPQRSTVWXYZ0-9]{26}$`,
+			},
+		},
+		{
+			name: "empty-writes",
+			input: &openfgav1.WriteRequest{
+				StoreId:              storeID,
+				AuthorizationModelId: modelID,
+				Writes:               &openfgav1.WriteRequestWrites{},
+			},
+			output: output{
+				errorCode:    codes.InvalidArgument,
+				errorMessage: `invalid WriteRequestWrites.TupleKeys: value must contain at least 1 item(s)`,
```

Comment:
Heh, I just noticed that this error message is ... sliiightly incorrect. It should say `either provide Writes or provide Deletes` ideally..

Reply:
So there seems to be a bit of nuance here. The validator alerts on the first validation error if a non-nil `Writes` or `Deletes` value is provided but if both are nil, the error is basically what you noted above. I made another case for this LMKWYT.